quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliPomIntegrityIT.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.ts.quarkus.cli;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import jakarta.inject.Inject;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.QuarkusCliClient;
+import io.quarkus.test.bootstrap.QuarkusCliRestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+
+@QuarkusScenario
+public class QuarkusCliPomIntegrityIT {
+    /*
+     * This scenario is related to the backport https://github.com/quarkusio/quarkus/issues/39088
+     *
+     */
+    private static final String NEW_EXTENSION = "quarkus-kafka-client";
+    private static final String COMMENT = "<!-- Disable native build on this module -->";
+
+    private static final String APP_NAME = "my-quarkus-app";
+
+    @Inject
+    static QuarkusCliClient cliClient;
+
+    @Test
+    public void shouldKeepCommentInPomAfterAddAndRemoveExtension() throws IOException {
+        final Path POM_PATH = Paths.get("target", QuarkusCliPomIntegrityIT.class.getSimpleName(), APP_NAME, "pom.xml");
```

Comment:
1. You shouldn't construct path to the file at service path manually in case the logic changes. FYI I have added a new method to TFW, so you can do:

```
QuarkusCliRestService app = cliClient.createApplication(APP_NAME);
var pomFile = app.getFileFromApplication("pom.xml");
```

2. non-static variables should not be upper case like `POM_PATH`, even though it's final, it should be camel